packages/asset/contracts/AssetCreate.sol

```diff
@@ -354,5 +356,267 @@ contract AssetCreate is
+        );
+
+    // A fee denoted in BPS that is charged when using the lazy minting feature, deducted from the creators payout
+    uint256 public lazyMintFee;
+    // The address that receives the lazy mint fee
+    address public lazyMintFeeReceiver;
+    // mapping of tokenId => maxSupply specified by the creator
+    mapping(uint256 => uint256) public availableToMint;
+
+    struct LazyMintData {
+        uint8 tier;
+        uint256 amount;
+        uint256 unitPrice;
+        address paymentToken;
+        string metadataHash;
+        uint256 maxSupply;
+        address creator;
+    }
+
+    struct LazyMintBatchData {
+        address[] creators;
+        uint8[] tiers;
+        uint256[] amounts;
+        uint256[] unitPrices;
+        address[] paymentTokens;
+        string[] metadataHashes;
+        uint256[] maxSupplies;
+    }
+
+    /// @notice Lazily creates a new asset with a signature
+    /// @dev Allows users to lazy mint assets
+    /// @param mintData The data for the lazy mint
+    function lazyCreateAsset(bytes memory signature, LazyMintData calldata mintData) external whenNotPaused {
+        require(
+            authValidator.verify(
+                signature,
+                _hashLazyMint(
+                    mintData.creator,
+                    signatureNonces[_msgSender()]++,
+                    mintData.tier,
+                    mintData.amount,
+                    mintData.unitPrice,
+                    mintData.paymentToken,
+                    mintData.metadataHash,
+                    mintData.maxSupply
+                )
+            ),
+            "AssetCreate: Invalid signature"
+        );
+        // check if asset has already been minted before
+        uint256 tokenId = assetContract.getTokenIdByMetadataHash(mintData.metadataHash);
+        if (tokenId == 0) {
+            tokenId = TokenIdUtils.generateTokenId(
+                mintData.creator,
+                mintData.tier,
+                ++creatorNonces[mintData.creator],
+                mintData.tier == 1 ? 1 : 0,
+                true
+            );
+            require(mintData.amount <= mintData.maxSupply, "AssetCreate: Max supply exceeded");
+            availableToMint[tokenId] = mintData.maxSupply - mintData.amount;
+        } else {
+            require(availableToMint[tokenId] >= mintData.amount, "AssetCreate: Max supply reached");
+            availableToMint[tokenId] -= mintData.amount;
+        }
+
+        // burn catalyst of a given tier
+        catalystContract.burnFrom(_msgSender(), mintData.tier, mintData.amount);
+        // send the payment to the creator after deducting the lazy mint fee
+        distributePayment(mintData.unitPrice, mintData.amount, mintData.paymentToken, mintData.creator);
+        // mint the asset
+        assetContract.mint(mintData.creator, tokenId, mintData.amount, mintData.metadataHash);
+        emit AssetLazyMinted(
```

Comment:
could be easier if we had same event for batch & simple ? 
Maybe we should ask backend point of view for this